quickfixj-core/src/main/java/quickfix/mina/ssl/SSLFilter.java

```diff
@@ -49,4 +59,26 @@ public void setEnabledCipherSuites(String[] cipherSuites) {
     public void setCipherSuites(String[] cipherSuites) {
         super.setEnabledCipherSuites(cipherSuites);
     }
+
+    @Override
+    public void onPreAdd(IoFilterChain parent, String name, NextFilter nextFilter)
+        throws SSLException {
+
+        if (useSNI) {

```

Comment:
This can be combined with  below if statement.

Reply:
Yes it can, but it will diminish the clarity of intent.

Reply:
I mean, the check on the enabling parameter is at a different "logical level" than the safety `instanceof` check, and mixing them may confuse code readers due to this. I can change if you insist.

Reply:
also, the `remoteAddress` local variable is introduced is introduced within the `if (useSNI) {` block, so to have it participate in the top-most condition would require a hacky style, e.g. `if (useSNI && ((remoteAddress = session.getRemoteAddress()) instanceof InetSocketAddress)) {`.
Similarly for the `session` local variable.

Reply:
I see the address. Keep it then.